src/tools/build-config

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+#  build-config
+#  Copyright 2017 Edward Wang <edward.c.wang@compdigitec.com>
+#
+#  Build the configuration database from a series of JSON config files.
+#  Dumps the output in JSON format to standard output.
+#  See README.config for more details.
+
+# pylint: disable=invalid-name
+
+import json
+import sys
+
+def unpack(config_dict, prefix=""):
+    """
+    Unpack the given config_dict, flattening key names recursively.
+    >>> p = unpack({"one": 1, "two": 2}, prefix="snack")
+    >>> p == {'snack.one': 1, 'snack.two': 2}
+    True
+    >>> p = unpack({"a": {"foo": 1, "bar": 2}})
+    >>> p == {'a.foo': 1, 'a.bar': 2}
+    True
+    >>> p = unpack({"a.b": {"foo": 1, "bar": 2}})
+    >>> p == {"a.b.foo": 1, "a.b.bar": 2}
+    True
+    >>> p = unpack({
+    ...     "a": {
+    ...         "foo": 1,
+    ...         "bar": 2
+    ...     },
+    ...     "b": {
+    ...         "baz": 3,
+    ...         "boom": {"rocket": "chip", "hwacha": "vector"}
+    ...     },
+    ... })
+    >>> p == {"a.foo": 1, "a.bar": 2, "b.baz": 3, "b.boom.rocket": "chip",
+    ...     "b.boom.hwacha": "vector"}
+    True
+    """
+    # We don't want an extra "." in the beginning.
+    real_prefix = "" if prefix == "" else prefix + "."
+    output_dict = {}
+    for key, value in config_dict.items():
+        if isinstance(value, dict):
+            output_dict.update(unpack(value, real_prefix + key))
+        else:
+            output_dict[real_prefix + key] = value
+    return output_dict
+
+def load_json_from_file(filename):
+    """
+    Load JSON from a filename, returning a blank dictionary if the file is
+    empty, instead of an error.
+    """
+    file_contents = open(filename, "r").read(10000000) # arbitrary limit
```

Comment:
In yaml2json below you use .read() I don't think this will be a common source of problem.